lasagne/layers/dnn.py

```diff
@@ -112,7 +112,13 @@ def get_output_for(self, input, **kwargs):
         # by default we assume 'cross', consistent with corrmm.
         conv_mode = 'conv' if self.flip_filters else 'cross'
 
-        image = T.alloc(0., *self.output_shape)
+        # Handle tensors with symbolic/unknown batch size by preserving
+        # the symbolic shape.
+        img_shape = list(self.output_shape)
+        if img_shape[0] is None:
+            img_shape[0] = input.shape[0]
+
+        image = tt.alloc(0., *img_shape)
```

Comment:
Oops, yes it should.

> On Apr 11, 2016, at 7:40 PM, ajbrock <notifications@github.com> wrote:
> 
> In lasagne/layers/dnn.py:
> 
> > @@ -112,7 +112,13 @@ def get_output_for(self, input, **kwargs):
> >          # by default we assume 'cross', consistent with corrmm.
> >          conv_mode = 'conv' if self.flip_filters else 'cross'
> >  
> > -        image = T.alloc(0., *self.output_shape)
> > +        # Handle tensors with symbolic/unknown batch size by preserving
> > +        # the symbolic shape.
> > +        img_shape = list(self.output_shape)
> > +        if img_shape[0] is None:
> > +            img_shape[0] = input.shape[0]
> > +
> > +        image = tt.alloc(0., *img_shape)
> Should this be T.alloc instead of tt.alloc?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>